Task4/train.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require_relative 'module_manufacturer.rb'
+require_relative 'instance_counter.rb'
+class Train
+  include Manufacturer
+  include InstanceCounter
+  attr_accessor :list_of_vagons, :type, :number, :speed
+  @@all_trains = Hash.new
+  NUMBER_FORMAT = /^([a-z]|[1-9]){3}-?([a-z]|[1-9]){2}$/.freeze # /\w{3}-?\w{2}/
+  def initialize(type, number)
+    @type = type
+    @number = number
+    @list_of_vagons = []
+    @speed = 0
+    @@all_trains[number] = self
+    register_instance
+    validate!
+  end
+
+  def validate?
```

Comment:
правильное название:
```suggestion
  def valid?
```

Reply:
этот метод абсолютно идентичный для всех классов
что можно сделать?